mavros/src/plugins/setpoint_attitude.cpp

```diff
@@ -215,6 +237,43 @@ class SetpointAttitudePlugin : public plugin::PluginBase,
 			send_attitude_ang_velocity(req->header.stamp, ang_vel, thrust_msg->thrust);
 	}
 
+	void rpyt_cb(const mav_msgs::RollPitchYawrateThrustConstPtr msg)
+	{
+		if(ignore_rpyt_messages_){
+			ROS_FATAL("Recieved roll_pitch_yaw_thrust_rate message, but ignore_rpyt_messages_ is true: "
+	 	        "the most likely cause of this is a failure to specify the thrust_scaling_factor,	"
+			"yaw_rate_scaling_factor or system_mass_kg parameters");
+		     	return;
+		}
+
+		// Set mask to ignore everything but thrust and attitude setpoint
+		uint8_t ignore_all_except_q_and_thrust = (7 << 0);
+
+		geometry_msgs::Quaternion orientation = 
+			tf::createQuaternionMsgFromRollPitchYaw( msg->roll, msg->pitch, 0);
+		// Transforms from thrust acceleration to a thrust force (scaling adjusts to the UAV
+		// propeller thrust, can be calculated from UAV calibration
+		double thrust = std::min(1.0, std::max(0.0, msg->thrust.z * thrust_scaling * system_mass_kg));
+		Eigen::Quaterniond desired_orientation;
+		Eigen::Vector3d body_rate;
+		tf::quaternionMsgToEigen(orientation, desired_orientation);
+		
+		// Transform desired orientation to represent aircraft->NED,
+		// MAVROS operates on orientation of base_link->ENU
+		auto ned_desired_orientation = ftf::transform_orientation_enu_ned(
+				ftf::transform_orientation_baselink_aircraft(desired_orientation));
+		
+		body_rate.x() = 0;
+		body_rate.y() = 0;
+		body_rate.z() = yaw_rate_scaling * ftf::detail::transform_frame_yaw(msg->yaw_rate);
```

Comment:
Why not:
```cpp
auto body_rate = Eigen::Vector(0, 0, yaw...);
```

Reply:
Without the scaling, you mean?

Reply:
Initializing, it looks a bit better than later fill zeros. Other possible code:

```
auto body_rate = Eigen::Vector3d::Zero();

body_rate.z() = yaw...;
```